cdap-security/src/main/java/co/cask/cdap/security/impersonation/DefaultImpersonator.java

```diff
@@ -36,6 +38,7 @@
 public class DefaultImpersonator implements Impersonator {
 
   private static final Logger LOG = LoggerFactory.getLogger(DefaultImpersonator.class);
+  private static final Logger IMPERSONATION_FAILTURE_LOG = Loggers.sampling(LOG, LogSamplers.limitRate(100));
```

Comment:
The rate limit frequency is in milliseconds. This mean it is allowed to emit one log per 100 ms. Is that what you wanted?

Reply:
No. I have updated the pr to emit one log per 100 calls.